File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    let answer = [];
    let score1 = 0;
    let score2 = 0;
    let score3 = 0;
    
    const sol1 = [1,2,3,4,5];
    const sol2 = [2,1,2,3,2,4,2,5];
    const sol3 = [3,3,1,1,2,2,4,4,5,5];
    
    for (var i = 0; i < answers.length; i++) {
        if (answers[i] === sol1[i % sol1.length])
            score1++;
        if (answers[i] === sol2[i % sol2.length])
            score2++;
        if (answers[i] === sol3[i % sol3.length])
            score3++
    }
    
    const maxScore = Math.max(score1, score2, score3);
    
    if (score1 === maxScore)
        answer.push(1);
    if (score2 === maxScore)
        answer.push(2);
    if (score3 === maxScore)
        answer.push(3);   
    
    return answer;
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    
    solution = [
        [1, 2, 3, 4, 5],
        [2, 1, 2, 3, 2, 4, 2, 5],
        [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    ]
    
    score = [0, 0, 0]
    
    
    for i in range(len(answers)):
        curr_answer = answers[i]
        for j in range(3):
            curr_solution = solution[j]
            if curr_solution[i % len(curr_solution)] == curr_answer:
                score[j] += 1
    
    max_score = max(score)
    
    for i in range(3):
        if score[i] == max_score:
            answer.append(i + 1)
    
    return answer